src/infrastructure/Listener.ts

```diff
@@ -114,58 +114,65 @@ export class Listener {
-                    } else if (message.block) {
-                        const networkType = parseInt(message.block.version.toString(16).substr(0, 2), 16);
-                        this.messageSubject.next({
-                            channelName: ListenerChannelName.block, message: new BlockInfo(
-                                message.meta.hash,
-                                message.meta.generationHash,
-                                message.meta.totalFee ? UInt64.fromNumericString(message.meta.totalFee) : new UInt64([0, 0]),
-                                message.meta.numTransactions,
-                                message.block.signature,
-                                PublicAccount.createFromPublicKey(message.block.signerPublicKey, networkType),
-                                networkType,
-                                parseInt(message.block.version.toString(16).substr(2, 2), 16), // Tx version
-                                message.block.type,
-                                UInt64.fromNumericString(message.block.height),
-                                UInt64.fromNumericString(message.block.timestamp),
-                                UInt64.fromNumericString(message.block.difficulty),
-                                message.block.feeMultiplier,
-                                message.block.previousBlockHash,
-                                message.block.blockTransactionsHash,
-                                message.block.blockReceiptsHash,
-                                message.block.stateHash,
-                                extractBeneficiary(message, networkType), // passing `message` as `blockDTO`
-                            ),
-                        });
-                    } else if (message.status) {
-                        this.messageSubject.next({
-                            channelName: ListenerChannelName.status, message: new TransactionStatusError(
-                                message.hash,
-                                message.status,
-                                Deadline.createFromDTO(message.deadline)),
-                        });
-                    } else if (message.parentHash) {
-                        this.messageSubject.next({
-                            channelName: ListenerChannelName.cosignature,
-                            message: new CosignatureSignedTransaction(message.parentHash, message.signature, message.signerPublicKey),
-                        });
-                    } else if (message.meta && message.meta.hash) {
-                        this.messageSubject.next({channelName: message.meta.channelName, message: message.meta.hash});
-                    }
+                    this.handleMessage(message, resolve);
                 };
             } else {
                 resolve();
             }
         });
     }
 
+    handleMessage(message: any, resolve) {
```

Comment:
Doc, make it `Private` and `@Internal`

Reply:
can it be called from the unit test if it's private?

Reply:
ah. okay, you added unit tests. so just need to make it `internal` then I think